src/components/ActivitiyForm/FormComponents/ImageUpload.tsx

```diff
@@ -0,0 +1,137 @@
+import { postActivityImage } from '@/libs/api/activities';
+import { PlusIcon, XIcon } from '@/libs/utils/Icon';
+import classNames from '@/libs/utils/classNames';
+import { useMutation } from '@tanstack/react-query';
+import Image from 'next/image';
+import { ChangeEvent, useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+interface ImageUploadProps {
+  name: string;
+  maxImages?: number;
+  label?: string;
+}
+
+export default function ImageUploader({
+  name,
+  maxImages = 1,
+  label = '이미지 등록',
+}: ImageUploadProps) {
+  const { setValue } = useFormContext();
+  const [imagePreviewUrls, setImagePreviewUrls] = useState<string[]>([]);
+
+  const imageUploadMutation = useMutation({
+    mutationFn: postActivityImage,
+    onSuccess: (data) => {
+      setImagePreviewUrls((prevUrls) => {
+        const updatedUrls = [...prevUrls, data.activityImageUrl].slice(
+          0,
+          maxImages,
+        );
+        return Array.from(new Set(updatedUrls));
+      });
+    },
+    onError: (error) => {
+      console.error('이미지 업로드 실패:', error);
+    },
+  });
+
+  useEffect(() => {
+    if (maxImages === 1) {
+      setValue(name, imagePreviewUrls[0] || '');
+    } else {
+      setValue(name, imagePreviewUrls);
+    }
+  }, [imagePreviewUrls, name, setValue, maxImages]);
+
+  const handleImageChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files;
+    if (files) {
+      for (let i = 0; i < files.length; i++) {
+        if (imagePreviewUrls.length >= maxImages) break;
+        await imageUploadMutation.mutateAsync(files[i]);
+      }
+    }
+  };
```

Comment:
```suggestion
  const handleImageChange = async (e) => {
    const files = event.target.files;
    if (files) {
      await Promise.all(files.map((file) => {
       imageUploadMutation.mutateAsync(file)
       })
    }
  };
```
이 부분은 map함수 사용 하면 병렬로 처리할 수 있을 것 같습니다!